ext/standard/basic_functions.c

```diff
@@ -1131,18 +1131,34 @@ PHP_FUNCTION(flush)
 /* {{{ Delay for a given number of seconds */
 PHP_FUNCTION(sleep)
 {
-	zend_long num;
+	zval *num;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_LONG(num)
+		Z_PARAM_NUMBER(num)
 	ZEND_PARSE_PARAMETERS_END();
-
-	if (num < 0) {
-		zend_argument_value_error(1, "must be greater than or equal to 0");
-		RETURN_THROWS();
-	}
-
-	RETURN_LONG(php_sleep((unsigned int)num));
+    if (Z_TYPE_P(num) == IS_DOUBLE) {
+		const double seconds = Z_DVAL_P(num);
+		if (UNEXPECTED(seconds < 0)) {
+			zend_argument_value_error(1, "must be greater than or equal to 0");
+			RETURN_THROWS();
+		}
+#ifdef HAVE_USLEEP
+		const unsigned int fraction_microseconds = (unsigned int)((seconds - (unsigned int)seconds) * 1000000);
+		if(fraction_microseconds > 0) {
+			usleep(fraction_microseconds);
+		}
+#endif
+		RETURN_LONG(php_sleep((unsigned int)seconds));
+	} else {
+		ZEND_ASSERT(Z_TYPE_P(num) == IS_LONG);
+		zend_long seconds = Z_LVAL_P(num);
+		if (UNEXPECTED(seconds < 0)) {
+			zend_argument_value_error(1, "must be greater than or equal to 0");
+			RETURN_THROWS();
+		}
+		RETURN_LONG(php_sleep((unsigned int)seconds));
+    }
+	ZEND_UNREACHABLE();
```

Comment:
this should be removed

Reply:
well it was accurate, but i can actually remove the else{} block entirely, done 👍